xbmc/FileItem.cpp

```diff
@@ -124,6 +124,10 @@ CFileItem::CFileItem(const CEpgInfoTagPtr& tag)
   Initialize();
 
   m_bIsFolder = false;
+  if (tag->HasPVRChannel())
+  {
+    m_pvrChannelInfoTag = tag->ChannelTag();
```

Comment:
I first tried to use a file item containing an epg tag, but there most of the code to properly display the replay is missing.

Thinking about it, I agree that the fileitem containg a pvrchannel should be used, but then we would need to tell the player the offset of the show such that it knows which EPG tag to display (instead of now), just like timeshift. I'm not sure if there already is a way to do this?
There comes an additional challenge: I'm not using the PVR addon as input stream but an inputstream addon (mpd) and as war as I can see, there is no way to let the player know about this offset?